### PR TITLE
Fix mapping TabPane title

### PR DESCRIPTION
**Pull request by taion**
Same as #1142.

### CODE REVIEW

**Comment by taion** [comment]
Don't merge yet - one more commit coming.

**Comment by taion** [comment]
Done - moved the deprecations to `componentWillMount` so they show up when rendering on server.

**Comment by taion** [comment]
One last docs fix - it's not `TabbedArea` and `TabPane` any more.

**Comment by jquense** [comment]
:shipit:

File: src/TabbedArea.js [hunk 1]
```diff
@@ -5,20 +5,23 @@ import ValidComponentChildren from './utils/ValidComponentChildren';
...
-    let {children, ...props} = this.props;
-    let tabTitles = [];
+    const {children, ...props} = this.props;
 
-    tabTitles = ValidComponentChildren.map(function(child) {
-      let {tab, ...others} = child.props;
-      tabTitles.push(<TabPane title={tab} {...others}/>);
+    const tabs = ValidComponentChildren.map(children, function (child) {
+      const {tab: title, ...others} = child.props;
+      return <TabPane title={title} {...others} />;
```
Comment (AlexKVal) [inline src/TabbedArea.js]:
```js
tab: title,
title={title}
```
I like it :+1:

**Pull request merged**